spec/requests/physical_storages_spec.rb

```diff
@@ -1,4 +1,31 @@
 describe "Physical Storages API" do
+  context "POST /api/physical_storages" do
+    it "creates new storage" do
+      api_basic_authorize(collection_action_identifier(:physical_storages, :create))
+      provider = FactoryBot.create(:ems_autosde)
+      request = {
+        "action"   => "create",
+        "resource" => {
+          "ems_id"                     => provider.id,
+          "name"                       => "test_storage",
+          "physical_storage_family_id" => "1",
+          "management_ip"              => "1.1.1.1",
+          "user"                       => "user",
+          "password"                   => "password"
+        }
+      }
+
+      post(api_physical_storages_url, :params => request)
+
+      expected = {
+        "success" => true,
+        "message" => "Creating Physical Storage",
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body['results'].first).to include(expected)
```

Comment:
You can replace above expected and the 2 expects with the following helper:

```
expect_single_action_result(:result => true, :message => /Creating Physical Storage test_storage/, :task => true)
```

Reply:
changed to expect_multiple_action_result, because create action is done on the collection_actions context